sympy/utilities/lambdify.py

```diff
@@ -907,8 +908,17 @@ def _lambdifygenerated(x):
     # mtime has to be None or else linecache.checkcache will remove it
     linecache.cache[filename] = (len(funcstr), None, funcstr.splitlines(True), filename) # type: ignore
```

Comment:
Fortunately the linecache doesn't actually store any references to the function objects themselves, only to some strings. So I expect this should be robust.